Code/GraphMol/FileParsers/MolFileWriter.cpp

```diff
@@ -50,12 +50,15 @@ int getQueryBondTopology(const Bond *bond) {
       qry->endChildren() - qry->beginChildren() == 2) {
     auto child1 = qry->beginChildren();
     auto child2 = child1 + 1;
-    if ((*child1)->getDescription() == "BondOr" &&
-        (*child2)->getDescription() == "BondInRing") {
-      qry = child2->get();
-    } else if ((*child1)->getDescription() == "BondInRing" &&
-               (*child2)->getDescription() == "BondOr") {
-      qry = child1->get();
+    if (((*child1)->getDescription() == "BondInRing") !=
```

Comment:
I'm not quite sure what the integer value for getQueryBondTopology means.

My guess:

```
0: no topology
1: query matches a bond in ring
2: query does not match a bond in ring
```

Why is this test only one level deep, couldn't this be in a lower tree?  It is unclear to me what BondOr or BondOrder goes to the deeper node.  I'm assuming this is a bit greedy and is specific to what can happen in an SDF file.

Reply:
Yeah, the int values are what one finds in a CTAB.

These specific tests are, as you guess, there to recognize particular things that happen in CTABs and some SMARTS queries.

Having a set of general predicates that allow us to recognize particularly query formulations would be a good thing. But that's a bigger change.